code/controllers/subsystems/ticker.dm

```diff
@@ -0,0 +1,537 @@
+				if(start_immediately)
+					fire()
+
+		if(GAME_STATE_SETTING_UP)
+			setup_failed = !setup()
+			if(setup_failed)
+				current_state = GAME_STATE_STARTUP
+				time_left = null
+				start_at = world.time + (CONFIG_GET(number/lobby_countdown) SECONDS)
+				start_immediately = FALSE
+				Master.SetRunLevel(RUNLEVEL_LOBBY)
+
+		if(GAME_STATE_PLAYING)
+			mode.process(wait * 0.1)
+
+			if(!roundend_check_paused && mode.check_finished(force_ending) || force_ending)
+				callHook("roundend")
+				current_state = GAME_STATE_FINISHED
+				config.ooc_allowed = TRUE
+				config.dooc_allowed = TRUE
+				declare_completion(force_ending)
+				addtimer(CALLBACK(src, .proc/Reboot), CONFIG_GET(number/vote_period) * 3 + 9 SECONDS)
+				Master.SetRunLevel(RUNLEVEL_POSTGAME)
+
+/datum/controller/subsystem/ticker/proc/setup()
+	to_chat(world, SPAN_BOLDNOTICE("<b>Enjoy the game!</b>"))
+	SEND_SOUND(world, sound(GLOB.using_map.welcome_sound))
+	var/init_start = world.timeofday
+	//Create and announce mode
+	if(GLOB.master_mode=="secret")
+		hide_mode = TRUE
+	else
+		hide_mode = FALSE
+
+	if(GLOB.master_mode=="secret")
+		if(secret_force_mode != "secret")
+			mode = config.pick_mode(secret_force_mode)
+		if(!mode)
+			mode = config.pick_mode(GLOB.master_mode)
+	else
+		mode = config.pick_mode(GLOB.master_mode)
+
+	CHECK_TICK
+	if(!mode.can_start(bypass_checks))
+		to_chat(world, "Reverting to pre-game lobby.")
+		QDEL_NULL(mode)
+		job_master.ResetOccupations()
+		return FALSE
+
+	CHECK_TICK
+	if(!mode.pre_setup() && !bypass_checks)
+		QDEL_NULL(mode)
+		to_chat(world, "<b>Error in pre-setup for [GLOB.master_mode].</b> Reverting to pre-game lobby.")
+		job_master.ResetOccupations()
+		return FALSE
+
+	CHECK_TICK
+	if(!mode.setup() && !bypass_checks)
+		QDEL_NULL(mode)
+		to_chat(world, "<b>Error in setup for [GLOB.master_mode].</b> Reverting to pre-game lobby.")
+		job_master.ResetOccupations()
+		return FALSE
+
+	callHook("roundstart")
+
+	CHECK_TICK
+	if(hide_mode)
+		to_chat(world, "<span class='infoplain'><B>The current game mode is - Secret!</B></span>")
+	else
+		mode.announce()
+
+	if(CONFIG_GET(flag/ooc_allowed))
+		config.ooc_allowed = TRUE
+
+	CHECK_TICK
+	//Here we will trigger the auto-observe and auto bst debug things
+	if (CONFIG_GET(flag/auto_observe))
+		for(var/client/C in GLOB.clients)
+			if (C.mob)
+				make_observer(C.mob)
+
+	CHECK_TICK
+	if (CONFIG_GET(flag/auto_bst))
+		for(var/client/C in GLOB.clients)
+			if (C.mob)
+				C.cmd_dev_bst(TRUE)
+
+	if (CONFIG_GET(flag/debug_verbs))
+		for(var/client/C in GLOB.clients)
+			C.enable_debug_verbs(TRUE)
+
+	CHECK_TICK
+	for(var/I in round_start_events)
+		var/datum/callback/cb = I
+		cb.InvokeAsync()
+	LAZYCLEARLIST(round_start_events)
+	CHECK_TICK
+
+	log_world("Game start took [(world.timeofday - init_start) / 10]s")
+	round_start_time = world.time
+
+	current_state = GAME_STATE_PLAYING
+	Master.SetRunLevel(RUNLEVEL_GAME)
+
+	CHECK_TICK
+	PostSetup()
+
+	if(CONFIG_GET(flag/sql_enabled))
```

Comment:
is this the same flag we had before? I seem to recall completely removing such a flag